seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java

```diff
@@ -109,24 +109,23 @@ public SeaTunnelDataType<SeaTunnelRow> getProducedType() {
         return this.physicalRowType;
     }
 
+    @Override
     public void deserialize(byte[] message, Collector<SeaTunnelRow> out) {
         if (message == null) {
             return;
         }
         ObjectNode jsonNode = (ObjectNode) convertBytes(message);
         assert jsonNode != null;
-        if (database != null) {
-            if (!databasePattern.matcher(jsonNode.get(FIELD_DATABASE).asText()).matches()) {
-                return;
-            }
+        if (database != null
+                && !databasePattern.matcher(jsonNode.get(FIELD_DATABASE).asText()).matches()) {
+            return;
         }
-        if (table != null) {
-            if (!tablePattern.matcher(jsonNode.get(FIELD_TABLE).asText()).matches()) {
-                return;
-            }
+        if (table != null && !tablePattern.matcher(jsonNode.get(FIELD_TABLE).asText()).matches()) {
+            return;
         }
         JsonNode dataNode = jsonNode.get(FIELD_DATA);
         String type = jsonNode.get(FIELD_TYPE).asText();
+        assert dataNode != null;
```

Comment:
Can a business exception be thrown?

Reply:
> 是否可以抛出业务异常？



> 是否可以抛出业务异常？
Like this?
if (dataNode == null) {
            new NullPointerException("deserialize cancal-json ‘data’ is null" + jsonNode)
        }

Reply:
> Can a business exception be thrown?

Like this?
`if (dataNode == null) {
new NullPointerException("deserialize cancal-json ‘data’ is null" + jsonNode)
}`

Reply:
Like this?
if (dataNode == null) { 
throw new NullPointerException("deserialize cancal-json ‘data’ is null" + jsonNode);
}

Reply:
[throw new SeaTunnelJsonFormatException()](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/exception/SeaTunnelJsonFormatException.java)

Reply:
please review

Reply:
> please review

<img width="1414" alt="image" src="https://user-images.githubusercontent.com/76689593/220889496-515524e8-5201-47aa-9770-b99e54fd68ee.png">